### ISSUE TITLE
7$: Error if await or yield is used in parameter list

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#241]
`await` and `yield` expressions are not allowed in function or arrow function parameter lists. Error if one is seen.

Example:
```js
async function f(x = await null) {}
```

**Comment by bruno-costanzo** [comment]
what are you looking for? that the function allow the await function or to parse it to avoid the error?

**Comment by strager** [comment]
> what are you looking for? that the function allow the await function or to parse it to avoid the error?

I want quick-lint-js to report an error. Currently quick-lint-js reports no error. Example of what I expect:
```javascript
async function f(x = await null) {}
                     ^^^^^ error: await is not allowed in parameter default values
```
`await` is already allowed inside the body of an `async` function. This task is about `await` specifically in the parameter list, which is disallowed per the language standard.
